src/cashflow.ts

```diff
@@ -9,32 +11,39 @@ const defaultRefillFactor = 20;
 const defaultRefillThresholdFactor = 8;
 
 /** The amount of tokens that will be sent to the user */
-export function creditAmount(token: TokenTicker): number {
+export function creditAmount(token: TokenTicker, factor: number = 1): Amount {
+  // Load this from connection?
+  const fractionalDigits = Number.parseInt(process.env.FAUCET_FRACTIONAL_DIGITS || "0", 10);
   const amountFromEnv = process.env[`FAUCET_CREDIT_AMOUNT_${token}`];
-  if (amountFromEnv) {
-    return Int53.fromString(amountFromEnv).toNumber();
-  } else {
-    return 10;
-  }
+  const wholeNumber = amountFromEnv ? Int53.fromString(amountFromEnv).toNumber() : 10;
+  const total = wholeNumber * factor;
+  // replace BN with BigInt with TypeScript 3.2 and node 11
+  const quantity = new BN(total).imul(new BN(10 ** fractionalDigits)).toString();
+
+  return {
+    quantity: quantity,
+    fractionalDigits: fractionalDigits,
+    tokenTicker: token,
+  };
 }
 
-export function refillAmount(token: TokenTicker): number {
-  const factorFromEnv = Number.parseInt(process.env.FAUCET_REFILL_FACTOR || "", 10) || undefined;
+export function refillAmount(token: TokenTicker): Amount {
+  const factorFromEnv = Number.parseInt(process.env.FAUCET_REFILL_FACTOR || "0", 10) || undefined;
```

Comment:
Thinking out loud here: this diff changes the bahavior:

```
> Number.parseInt("", 10)
NaN
> Number.parseInt("0", 10)
0
```

However, both values are falsy (which I did not realize when I wrote this code), so it is impossible to set FAUCET_REFILL_FACTOR=0. Which is not a problem. All 👍